bson/encode.go

```diff
@@ -229,7 +229,12 @@ func (e *encoder) addStruct(v reflect.Value) {
 		if info.Inline == nil {
 			value = v.Field(info.Num)
 		} else {
-			value = v.FieldByIndex(info.Inline)
+			field, errField := safeFieldByIndex(v, info.Inline)
```

Comment:
this change will silently skip encoding errors, where they used to panic before.
Not very familiar with the bson encoder anymore, @domodwyer thoughts?

Reply:
Encoder has a `structInfo` object, based only on type, not on value.
As we allow pointers to structs here, we can have chain of inline indices and one of elements behind a `n-1` index might be need, so trying to get element by index `n` will panic because of `reflect: indirection through nil pointer to embedded struct`

Reply:
I think this is probably fine either way - relying on the encoder to panic because the input was invalid is a bit of a stretch in terms of backwards-compatibility.